addons/sale_expense/models/product_template.py

```diff
@@ -11,10 +11,11 @@ def _default_visible_expense_policy(self):
         visibility = self.user_has_groups('hr_expense.group_hr_expense_user')
         return visibility or super(ProductTemplate, self)._default_visible_expense_policy()
 
+    @api.depends('can_be_expensed')
     def _compute_visible_expense_policy(self):
         super(ProductTemplate, self)._compute_visible_expense_policy()
 
-        visibility = self.user_has_groups('hr_expense.group_hr_expense_user')
+        visibility = self.user_has_groups('hr_expense.group_hr_expense_user') and self.can_be_expensed
```

Comment:
Hi @laurent-brisbois 

Sorry to bother you but the result is not the same :)

Before: Invisible IF not ``visible_expense_policy`` OR not ``can_be_expensed``

Now: Invisible IF not ``visible_expense_policy``
Knowing that: ``visible_expense_policy`` is True IF user has_group ``group_analytic_accounting`` OR (``group_expense_user`` AND ``can_be_expensed``) OR ``group_project_user``

So it means that if I'm a project user, I can see the radio buttons, whatever the value of ``can_be_expensed``.

If we want the ``can_be_expensed`` value to be exclusive concerning the visibility of the radio button, we could do something like:
```python3
    def _compute_visible_expense_policy(self):
        expense_products = self.filtered(lambda p: p.can_be_expensed)
        for product_template in self - expense_products:
            product_template.visible_expense_policy = False

        super(ProductTemplate, expense_products)._compute_visible_expense_policy()
        visibility = self.user_has_groups('hr_expense.group_hr_expense_user')
        for product_template in expense_products:
            if not product_template.visible_expense_policy:
                product_template.visible_expense_policy = visibility
```

Could you check and tell me what you think about that ? 

Thanks.

Reply:
Hey @tivisse 
Considering :
A == having the group
B == can_be_expensed = true
It was `invisible` if ~A OR ~B, which means "not having the group OR can_be_expensed = false". Cases : (1 0), (0 0), (0 1).
However, here we are talking about `visibility`, which is the opposite of `invisible`.
So ~(~A OR ~B) = A AND B (which is equal to the remaining case (1 1).

Am I wrong ?

Reply:
Oh but there are other groups to consider, my bad, I'll check that.

Reply:
Tested all possible cases and it works. So I pushed that.